tests/unit/parts/test_lifecycle.py

```diff
@@ -1860,3 +1860,39 @@ def test_lifecycle_write_metadata(
             primed_stage_packages=[],
         )
     ]
+
+
+@pytest.mark.parametrize(
+    "base,build_base,grade,allow_return_devel,expected_base",
+    [
+        ["coreXX", None, "stable", False, "coreXX"],
+        ["coreXX", None, "stable", True, "coreXX"],
+        ["coreXX", None, "devel", False, "coreXX"],
+        ["coreXX", None, "devel", True, "coreXX"],
+        ["coreXX", "coreYY", "stable", False, "coreYY"],
+        ["coreXX", "coreYY", "stable", True, "coreYY"],
+        ["coreXX", "coreYY", "devel", False, "coreYY"],
+        ["coreXX", "coreYY", "devel", True, "coreYY"],
+        ["coreXX", "devel", "devel", False, "coreXX"],
+        ["coreXX", "devel", "devel", True, "devel"],
+    ],
+)
+def test_lifecycle_get_effective_base(
+    emitter,
+    mocker,
+    snapcraft_yaml,
+    tmp_path,
+    base,
+    build_base,
+    grade,
+    allow_return_devel,
+    expected_base,
+):
+    """Get the effective base from the project."""
```

Comment:
Should this be moved to `tests/unit/test_projects.py`?

Reply:
Yeah, this was not the case in the initial test, so I put it here. It can indeed be moved into `test_projects.py`